src/main/java/uk/gov/hmcts/ethos/replacement/docmosis/service/Et3ResponseService.java

```diff
@@ -185,4 +204,156 @@ private static String getRespondentNames(CaseData caseData) {
             .map(o -> o.getValue().getRespondentName())
             .collect(Collectors.joining(", "));
     }
+
+    /**
+     * Validates the representation of respondents in the given case data and extracts
+     * contact details of the first represented respondent's representative.
+     *
+     * <p>This method performs the following steps:
+     * <ul>
+     *   <li>Attempts to retrieve the list of respondent indexes that are marked as represented
+     *       using the provided user token and case ID.</li>
+     *   <li>If an error occurs during this retrieval (e.g. service exception or I/O issue), the
+     *       error message is added to the returned list.</li>
+     *   <li>If no represented respondents are found, an appropriate error message is added.</li>
+     *   <li>If at least one represented respondent is found, their representative's phone number
+     *       and address are extracted and stored in the case data as ET3 response details.</li>
+     * </ul>
+     *
+     * @param userToken the user authentication token used to retrieve represented respondent indexes.
+     * @param caseData the case data containing respondent and representative information.
+     * @return a list of error messages if any issues are encountered during validation.
+     */
+    public List<String> validateAndExtractRepresentativeContact(String userToken, CaseData caseData) {
+        List<Integer> representedRespondentIndexes = new ArrayList<>();
+        List<String> errors = new ArrayList<>();
+        try {
+            representedRespondentIndexes = getRepresentedRespondentIndexes(userToken, caseData.getCcdID());
+        } catch (GenericServiceException | IOException ex) {
+            errors.add(ex.getMessage());
+        }
+        if (CollectionUtils.isEmpty(representedRespondentIndexes) || representedRespondentIndexes.getFirst() == null) {
+            errors.add(NO_REPRESENTED_RESPONDENT_FOUND);
+        }
+        RepresentedTypeR representedTypeR = caseData.getRepCollection()
+                .get(representedRespondentIndexes.get(representedRespondentIndexes.getFirst())).getValue();
+        caseData.setEt3ResponsePhone(representedTypeR.getRepresentativePhoneNumber());
+        caseData.setEt3ResponseAddress(representedTypeR.getRepresentativeAddress());
+        return errors;
+    }
```

Comment:
Would it be better to log this error message rather than add it to the errors that we show an end user?